src/main/java/com/gmail/inverseconduit/bot/DefaultBot.java

```diff
@@ -52,6 +53,13 @@ private void processMessageQueue() {
     }
 
     private void processMessage(final ChatMessage chatMessage) {
+    	for (final CommandHandle listener : listeners){
+    		final String response = listener.execute(chatMessage);
+    		if (response != null){
```

Comment:
what about empty responses? You could drastically shorten this construct with java 8:

    listeners.stream().map(l -> l.execute(chatMessage)).findFirst().ifPresent(chatInterface.sendMessage(...));